core/src/either.rs

```diff
@@ -101,76 +84,93 @@ where
     A: AsyncWrite,
     B: AsyncWrite,
 {
-    fn shutdown(&mut self) -> Poll<(), IoError> {
-        match self {
-            EitherOutput::First(a) => a.shutdown(),
-            EitherOutput::Second(b) => b.shutdown(),
+    #[project]
+    fn poll_write(self: Pin<&mut Self>, cx: &mut Context, buf: &[u8]) -> Poll<Result<usize, IoError>> {
```

Comment:
We should also override [`AsyncWrite::poll_write_vectored`](https://docs.rs/futures/0.3.1/futures/prelude/trait.AsyncWrite.html#method.poll_write_vectored).